### PR TITLE
Use TLS 1.2 by default when min_version is not defined

### PR DESCRIPTION
**Pull request by rapphil**
**Description:** 

Use TLS 1.2 by default when min_version is not defined. 

TLS 1.0 and 1.1 are widely deprecated. It does not make sense to offer users with an insecure default.

This PR changes the default TLS min_version for both server and client to 1.2. This is reverting part of a recent change: https://github.com/open-telemetry/opentelemetry-collector/pull/5480

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/5627

**Testing:** Unit tests were modified to account for this case.

**Documentation:** Documentation was updated describing the new default.

### CODE REVIEW

**Review by jpkrohling — COMMENTED** [review]
I'm all for this change, which has been the case in the past, but a recent change made our code relay the default to go's crypto/tls package. Would you be able to dig that issue and link it here? A git blame on the "// Defaults will ..." comment should reveal the PR/issue that changed that.

**Review by Aneurysm9 — APPROVED** [review]
Please add a `CHANGELOG.md` entry.

**Comment by rapphil** [comment]
> I'm all for this change, which has been the case in the past, but a recent change made our code relay the default to go's crypto/tls package. Would you be able to dig that issue and link it here? A git blame on the "// Defaults will ..." comment should reveal the PR/issue that changed that.

I added the previous PR in the description.

**Comment by jpkrohling** [comment]
@bogdandrutu, in the previous PR, you seemed to favor delegating the default protocol to the underlying library. I believe setting 1.2 as the min version is the right thing to do here, despite the underlying library setting a lower min version, but it would be good to have your OK on this before moving forward.

**Comment by bogdandrutu** [comment]
Usually I prefer to be consistent with the standard library. Not an expert in security, and not sure what it means to use the deprecated version.

**Comment by rapphil** [comment]
> Usually I prefer to be consistent with the standard library. Not an expert in security, and not sure what it means to use the deprecated version.

I agree with your approach, but I think it is not applicable in this case because promoting security best practices with secure defaults has higher priority. 

I'm not a security expert either but I think we can trust open standards to guide us here:
https://datatracker.ietf.org/doc/rfc8996/

> Removing support for older versions from implementations reduces the
   attack surface, reduces opportunity for misconfiguration, and
   streamlines library and product maintenance.

The RFC has references for possible attacks for which TLS1.0 and 1.1 are vulnerable to.

**Comment by jpkrohling** [comment]
In security circles, people just assume TLS 1.0 and 1.1 are effectively broken. There might be people out there with good reasons to use those versions, so I wouldn't artificially limit its usage, but I would definitely have 1.2 as the default min version.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 - Replace `processorhelper.New[Traces|Metrics|Logs]Exporter` with `processorhelper.New[Traces|Metrics|Logs]ProcessorWithCreateSettings` definition (#5915)
 - Replace `exporterhelper.New[Traces|Metrics|Logs]Exporter` with `exporterhelper.New[Traces|Metrics|Logs]ExporterWithContext` definition (#5914)
 - Replace ``component.NewExtensionFactory`` with `component.NewExtensionFactoryWithStabilityLevel` definition (#5917)
+- Set TLS 1.2 as default for `min_version` for TLS configuration in clients and servers in case this property is not defined (#5956)
```
Comment (bogdandrutu) [inline CHANGELOG.md:22]: The statement is partially incorrect, since for client it was already 1.2

Comment (rapphil) [reply CHANGELOG.md:22]: It is setting explicitly, no? Before it was implicit.

Comment (bogdandrutu) [reply CHANGELOG.md:22]: But the action is not a "breaking change" since the result for the user is the same.

Comment (rapphil) [reply CHANGELOG.md]: Good point. I updated the changelog.

**Comment by rapphil** [comment]
@bogdandrutu @jpkrohling 
Is there anything else that I can do to get your approvals?

**Pull request merged**